src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/util/BlackDuckResponseCache.java

```diff
@@ -145,19 +146,9 @@ public String getSeverity(String vulnerabilityUrl) {
     }
 
     // TODO update this code with an Object from blackduck-common-api when available
-    private Optional<String> getCvss3Severity(VulnerabilityView vulnerabilityView) {
-        Boolean useCvss3 = vulnerabilityView.getUseCvss3();
-        if (null != useCvss3 && useCvss3) {
-            JsonObject vulnJsonObject = vulnerabilityView.getJsonElement().getAsJsonObject();
-            JsonElement cvss3 = vulnJsonObject.get("cvss3");
-            if (null != cvss3) {
-                JsonElement cvss3Severity = cvss3.getAsJsonObject().get("severity");
-                if (null != cvss3Severity) {
-                    return Optional.of(cvss3Severity.getAsString());
-                }
-            }
-        }
-        return Optional.empty();
+    private Optional<ProjectVersionVulnerableBomComponentsItemsVulnerabilityWithRemediationSeverityType> getCvss3Severity(VulnerabilityView vulnerabilityView) {
+        ProjectVersionVulnerableBomComponentsItemsVulnerabilityWithRemediationSeverityType severity = vulnerabilityView.getCvss3().getSeverity();
+        return Optional.of(severity);
```

Comment:
Can you change this to `Optional.ofNullable(severity)` ? Just in case the severity is null

Reply:
certainly

Reply:
We should probably also check that `getCvss3()` is not null. Alternatively we can keep looking at the `useCvss3` flag in the object.